bitseq/sequence.go

```diff
@@ -308,17 +308,17 @@ func (h *Handle) set(ordinal, start, end uint64, any bool, release bool) (uint64
 
 	for {
 		var store datastore.DataStore
-		h.Lock()
+		h.RLock()
 		store = h.store
-		h.Unlock()
+		h.RUnlock()
 		if store != nil {
-			if err := store.GetObject(datastore.Key(h.Key()...), h); err != nil && err != datastore.ErrKeyNotFound {
+			if err = store.GetObject(datastore.Key(h.Key()...), h); err != nil && err != datastore.ErrKeyNotFound {
 				return ret, err
 			}
 		}
 
-		h.Lock()
 		// Get position if available
+		h.Lock()
```

Comment:
eyesore. pls move it above the comment to avoid unnecessary changes in the diff.